pkg/registry/core/pod/rest/subresources.go

```diff
@@ -213,8 +215,14 @@ func (r *PortForwardREST) Connect(ctx context.Context, name string, opts runtime
 	return newThrottledUpgradeAwareProxyHandler(location, transport, false, true, responder), nil
 }
 
-func newThrottledUpgradeAwareProxyHandler(location *url.URL, transport http.RoundTripper, wrapTransport, upgradeRequired bool, responder rest.Responder) *proxy.UpgradeAwareHandler {
+func newThrottledUpgradeAwareProxyHandler(location *url.URL, transport http.RoundTripper, wrapTransport, upgradeRequired bool, responder rest.Responder) http.Handler {
 	handler := proxy.NewUpgradeAwareHandler(location, transport, wrapTransport, upgradeRequired, proxy.NewErrorResponder(responder))
 	handler.MaxBytesPerSec = capabilities.Get().PerConnectionBandwidthLimitBytesPerSec
+	if utilfeature.DefaultFeatureGate.Enabled(features.ClientRemoteCommandWebsockets) {
```

Comment:
newThrottledUpgradeAwareProxyHandler is used for more than exec/attach... put the logic to do this wrapping in AttachREST#Connect and ExecREST#Connect instead

Reply:
Fixed. Moved wrapper into only `AttachREST` and `ExecREST`. Updated `StreamTranslator` to take the `Options` describing the requested streams. Also validates the requested streams here.